### PR TITLE
Update manifest breaking change

### PR DESCRIPTION
**Pull request by gewarren**
I think the previous way this was written, it included a bunch of (useful) information that wasn't directly related to the breaking change. It wasn't clear what exactly the breaking change was, also because of some misinformation in the doc. This PR strips out the unrelated information. I'll add a new, separate article that details how manifest file names are chosen in #18304.

Also adds this to the .NET Framework -> .NET Core breaking changes page.

[Internal preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/2.2-3.0?branch=pr-en-us-18306#resource-manifest-file-name-change).

Contributes to #16964.

### CODE REVIEW

**Comment by gewarren** [comment]
@IEvangelist Ping for review when you get a chance. The warnings are standard for include files.

File: includes/core-changes/msbuild/3.0/resource-manifest-name.md [hunk 1]
```diff
@@ -1,78 +1,35 @@
...
+Starting in .NET Core 3.0, if no `LogicalName`, `ManifestResourceName`, or `DependentUpon` metadata is specified for an `EmbeddedResource` item in the project file, and if the resource file is colocated with a source file of the same name (for example, *Form1.resx* and *Form1.cs*), MSBuild generates a manifest file name in the pattern `<Namespace>.<ClassName>.resources`, where the namespace and class name are extracted from the first type in the colocated source file. For example, the generated manifest name for a resource file named *Form1.resx* that's colocated with a source file named *Form1.cs* is *MyNamespace.Form1.resources*. The key thing to note is that the first part of the file name is different (*MyNamespace* instead of *MyProject*).
```
Comment (IEvangelist) [inline includes/core-changes/msbuild/3.0/resource-manifest-name.md:13]: The first sentence here is really long, could you split it up? Also, should `colocated` be hyphenated, `co-located`?

Comment (gewarren) [reply includes/core-changes/msbuild/3.0/resource-manifest-name.md]: The [style guide](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=28765) says not to hyphenate words with prefixes unless omitting the hyphen could confuse the reader.

**Review by IEvangelist — APPROVED** [review]
Looks good to me :shipit:

**Comment by gewarren** [comment]
Thanks David!

**Review by benvillalobos — COMMENTED** [review]
Late review, but LGTM!

**Comment by gewarren** [comment]
Thanks Ben!

**Pull request merged**